### PR TITLE
v2beta release, using google life-sciences api

### PR DESCRIPTION
**Pull request by alborzml**
version v2beta, using google life-sciences api

### CODE REVIEW

File: pipelines/main.go [hunk 1]
```diff
@@ -98,7 +103,7 @@ func newService(ctx context.Context, basePath string) (*genomics.Service, error)
 
 	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: &transport})
 
-	client, err := google.DefaultClient(ctx, genomics.GenomicsScope)
+	client, err := google.DefaultClient(ctx, genomicsScope)
```
Comment (kemp-google) [inline pipelines/main.go]: You should still use the scope constant from the generated package (eg v2beta.CloudPlatformScope)

File: pipelines/internal/common/common.go [hunk 2]
```diff
@@ -7,22 +7,39 @@ import (
...
+				name = path.Join("operations/", name)
+			}
+			name = path.Join("locations", location, name)
 		}
 		name = path.Join("projects", project, name)
 	}
 	return name
 }
 
+//func ExpandOperationName(prefixes, values []string, name string) (string, error) {
```
Comment (kemp-google) [inline pipelines/internal/common/common.go]: Remove the commented out block

File: pipelines/internal/commands/watch/watch.go [hunk 3]
```diff
@@ -84,7 +85,11 @@ func watch(ctx context.Context, service *genomics.Service, name string) (interfa
 				fmt.Println(timestamp.Format("15:04:05"), metadata.Events[i].Description)
 
 				if *details {
-					fmt.Println(string(metadata.Events[i].Details))
+					encoded, err := json.MarshalIndent(metadata.Events[i], "", " ")
+					if err != nil {
+						return nil, fmt.Errorf("encoding event: %v", err)
+					}
+					fmt.Printf("%s\n", encoded)
```
Comment (kemp-google) [inline pipelines/internal/commands/watch/watch.go]: Use Println instead of %s\n [as it was before]

File: pipelines/internal/commands/watch/watch.go [hunk 4]
```diff
@@ -58,8 +58,9 @@ func watch(ctx context.Context, service *genomics.Service, name string) (interfa
 	var events []*genomics.Event
 	const initialDelay = 5 * time.Second
 	delay := initialDelay
+	fmt.Println(name)
```
Comment (kemp-google) [inline pipelines/internal/commands/watch/watch.go]: This looks like an output change?

File: pipelines/internal/commands/watch/watch.go [hunk 5]
```diff
@@ -34,13 +34,13 @@ var (
 	details = flags.Bool("details", false, "show event details")
 )
 
-func Invoke(ctx context.Context, service *genomics.Service, project string, arguments []string) error {
+func Invoke(ctx context.Context, service *genomics.Service, project string, location string, arguments []string) error {
```
Comment (kemp-google) [inline pipelines/internal/commands/watch/watch.go]: project, location string

File: pipelines/internal/commands/run/run.go [hunk 6]
```diff
@@ -394,17 +395,17 @@ func buildRequest(filename, project string) (*genomics.RunPipelineRequest, error
...
 		if err != nil {
 			return nil, fmt.Errorf("expanding zones: %v", err)
 		}
 		resources.Zones = zones
-    }
-    if len(resources.Zones) + len(resources.Regions) == 0 {
-        resources.Zones = []string{"us-east1-d"}
-    }
+	}
+	if len(resources.Zones)+len(resources.Regions) == 0 {
```
Comment (kemp-google) [inline pipelines/internal/commands/run/run.go]: These look like unrelated formatting changes.  They should be submitted separately

File: pipelines/internal/commands/run/run.go [hunk 7]
```diff
@@ -207,18 +207,20 @@ func Invoke(ctx context.Context, service *genomics.Service, project string, argu
 		return nil
 	}
 
-	return runPipeline(ctx, service, req)
+	return runPipeline(ctx, service, req, project, location)
 }
 
-func runPipeline(ctx context.Context, service *genomics.Service, req *genomics.RunPipelineRequest) error {
+func runPipeline(ctx context.Context, service *genomics.Service, req *genomics.RunPipelineRequest, project string, location string) error {
```
Comment (kemp-google) [inline pipelines/internal/commands/run/run.go]: project, location string

File: pipelines/internal/commands/run/run.go [hunk 8]
```diff
@@ -181,7 +181,7 @@ func init() {
 	flags.Var(&common.MapFlagValue{vmLabels}, "vm-labels", "label names and values to apply to the virtual machine")
 }
 
-func Invoke(ctx context.Context, service *genomics.Service, project string, arguments []string) error {
+func Invoke(ctx context.Context, service *genomics.Service, project string, location string, arguments []string) error {
```
Comment (kemp-google) [inline pipelines/internal/commands/run/run.go]: project, location string

File: pipelines/internal/commands/run/run.go (hunk 6 above)
Comment (alborzml) [reply pipelines/internal/commands/run/run.go]: Removed changes

File: pipelines/internal/commands/query/query.go [hunk 9]
```diff
@@ -32,11 +32,11 @@ var (
 	all    = flags.Bool("all", false, "show all operations (when false, show only running operations)")
 )
 
-func Invoke(ctx context.Context, service *genomics.Service, project string, arguments []string) error {
+func Invoke(ctx context.Context, service *genomics.Service, project, location string, arguments []string) error {
 	flags.Parse(arguments)
 
-	path := fmt.Sprintf("projects/%s/operations", project)
-	call := service.Projects.Operations.List(path).Context(ctx)
+	path := fmt.Sprintf("projects/%s/locations/%s/operations", project, location)
```
Comment (tjennison-google) [inline pipelines/internal/commands/query/query.go]: Use path.Join here as in run.go

File: pipelines/internal/commands/run/run.go [hunk 10]
```diff
@@ -494,12 +495,21 @@ func parse(line string) (*genomics.Action, error) {
...
+	}
 	if n := strings.Index(line, "#"); n >= 0 {
 		for _, option := range strings.Fields(strings.TrimSpace(line[n+1:])) {
 			if n := strings.Index(option, "="); n >= 0 {
 				options[option[:n]] = option[n+1:]
-			} else {
-				action.Flags = append(action.Flags, strings.ToUpper(option))
+			} else if p, ok := flags[strings.ToUpper(option)]; ok {
+				*p = true
 			}
```
Comment (tjennison-google) [inline pipelines/internal/commands/run/run.go]: Add an else case that returns an error so incorrect flags aren't silently ignored.

File: pipelines/internal/commands/query/query.go (hunk 9 above)
Comment (alborzml) [reply pipelines/internal/commands/query/query.go]: Done

File: pipelines/internal/commands/run/run.go (hunk 10 above)
Comment (alborzml) [reply pipelines/internal/commands/run/run.go]: Done

File: pipelines/internal/commands/run/run.go [hunk 11]
```diff
@@ -771,10 +783,11 @@ func gcsFuse(buckets map[string]string) []*genomics.Action {
 	var actions []*genomics.Action
 	for bucket, path := range buckets {
 		actions = append(actions, &genomics.Action{
-			ImageUri: "gcr.io/cloud-genomics-pipelines/gcsfuse",
-			Commands: []string{"--implicit-dirs", "--foreground", bucket, path},
-			Flags:    []string{"ENABLE_FUSE", "RUN_IN_BACKGROUND"},
-			Mounts:   []*genomics.Mount{googleRoot},
+			ImageUri:        "gcr.io/cloud-genomics-pipelines/gcsfuse",
```
Comment (kemp-google) [inline pipelines/internal/commands/run/run.go]: unrelated formatting change?
